src/couch/priv/couch_js/60/main.cpp

```diff
@@ -473,6 +473,7 @@ main(int argc, const char* argv[])
         // Compile and run
         JS::CompileOptions options(cx);
         options.setFileAndLine(args->scripts[i], 1);
+        options.setUTF8(true);
```

Comment:
Good catch!

In SM68 (I'm working on this PR right now, hope to have it done today) this conversion is done explicitly via the (`JS::SourceText` class)[https://searchfox.org/mozilla-central/source/js/public/SourceText.h#29-44] and associated JS functions to convert to UTF-8 or UTF-16 "code units."

Reply:
heh, yeah, I just stumbled over it. And also tested to make sure it's actually necessary and effective. Hooray for scouring the source and headers for SpiderMonkey.